specs/driver_spec.rb

```diff
@@ -0,0 +1,182 @@
+
+
+  describe "#self.all" do
+
+    it "Returns an array." do
+      path = "./support/drivers_spec1.csv"
+      all_drivers = RideSharing::Driver.all(path)
+      all_drivers.must_be_kind_of Array
+    end
+
+    it "The elements in the returned array must be RideSharing::Driver instances " do
+      path = "./support/drivers_spec1.csv"
+      all_drivers = RideSharing::Driver.all(path)
+      all_drivers.each do |obj|
+        obj.must_be_kind_of RideSharing::Driver
+      end
+      all_drivers.length.must_equal 3
+    end
+
+    it "Does not initialize driver for any of the invalid inputs in /support/drivers_spec2.csv" do
+      path = "./support/drivers_spec2.csv"
+      all_drivers = RideSharing::Driver.all(path)
+      all_drivers.length.must_equal 14 # out of a total of 20 inputs in support/drivers_spec2.csv
+      all_drivers[0].name.must_equal "Bernardo Prosacco"
+      all_drivers[1].name.wont_equal "Emory Rosenbaum"
+    end
+
+    it "drivers.csv should return 100 drivers" do
+      all_drivers = RideSharing::Driver.all
+      all_drivers.length.must_equal 100
+    end
+  end # End of describe "Driver#self.all"
+
+
+  describe "#self.find" do
+    it "returns an object of class RideSharing::Driver if the id exists" do
+      returned_object = RideSharing::Driver.find(1)
+      returned_object.must_be_kind_of RideSharing::Driver
+    end
+
+    it "returns right name" do
+      returned_object = RideSharing::Driver.find(1)
+      returned_object.name.must_equal "Bernardo Prosacco"
+      returned_object = RideSharing::Driver.find(10)
+      returned_object.name.must_equal "Dr. Kenton Berge"
+    end
+
+    # it "Throws an error if id is not found" do
+    #   proc{
+    #     RideSharing::Driver.find(101)
+    #   }.must_raise ArgumentError
+    # end
+    it "Returns nil if id is not found" do
+      RideSharing::Driver.find(0).must_be_nil
+      RideSharing::Driver.find(101).must_be_nil
+    end
+  end # End of describe "Driver#self.find"
+
+  describe "#list_of_trips" do
+    # This method is solely dependent upon a class method in the class Trip.
+    # Blind trust is giving that no tests should be needed. The method used should have been thouroughly tested in the class Trip.
+  end
+
+
+  describe "#average_rating" do
+    # let(:driver1) {RideSharing::Driver.new(1, "Bernardo Prosacco", "WBWSS52P9NEYLVDE9")}
+    let(:driver1) {RideSharing::Driver.find(1)}
+    let(:driver100) {RideSharing::Driver.find(100)}
+
+    it "Return value should be a float" do
+      driver1.average_rating.must_be_kind_of Float
+    end
+
+    it "Test of calculation of the average of the rating with number of rides > 0  " do
+      sum_of_ratings = driver1.list_of_trips.map { |trip| trip.rating}.sum.to_f
+      driver1.average_rating.must_equal sum_of_ratings/9
+    end
+
+    it "Test of calculation of the average of the rating with number of rides == 0  " do
+      driver100 = RideSharing::Driver.find(100)
```

Comment:
Because you defined `driver100` in the let block above, you don't need to re-find it here.